web/client/components/shapefile/ShapefileUploadAndStyle.jsx

```diff
@@ -58,10 +58,38 @@ class ShapeFileUploadAndStyle extends React.Component {
 
     static defaultProps = {
         shapeLoading: () => {},
-        readFiles: files => files.map((file) => {
-            return FileUtils.readZip(file).then((buffer) => {
-                return FileUtils.shpToGeoJSON(buffer);
-            });
+        readFiles: (files) => files.map((file) => {
+            const ext = FileUtils.recognizeExt(file.name);
+            const type = FileUtils.MIME_LOOKUPS[ext];
+            if (type === 'application/vnd.google-earth.kml+xml') {
+                return FileUtils.readKml(file).then((xml) => {
+                    return FileUtils.kmlToGeoJSON(xml);
+                }).catch((e) => {
+                    return e.message;
```

Comment:
Is this used to show an error to the user at all?

Reply:
This is an additional catch, the error is catched and show here:
https://github.com/mricca/MapStore2/blob/a00b1a4063fcc81b03bd929a8d268135f7998041/web/client/components/shapefile/ShapefileUploadAndStyle.jsx#L207

what is the most logical thing to do?

Reply:
Uhm... if it's already catched elsewhere, it's not so useful to have another catch here

Reply:
ok, I will remove it